apps/api/src/app/shared/services/query-parser/query-validator.service.ts

```diff
@@ -11,13 +11,33 @@ type QueryIssue = {
 export enum QueryIssueTypeEnum {
   INVALID_STRUCTURE = 'INVALID_STRUCTURE',
   MISSING_VALUE = 'MISSING_VALUE',
+  INVALID_FIELD_VALUE = 'INVALID_FIELD_VALUE',
 }
 
 export class QueryValidatorService {
+  constructor(
+    private allowedFields: string[],
+    private allowedPrefixes: string[]
+  ) {}
+
   private isInvalidFieldReference(field: unknown) {
     return !field || typeof field !== 'object' || !('var' in field);
   }
 
+  private isInvalidFieldValue(field: unknown) {
+    if (this.isInvalidFieldReference(field)) {
+      return true;
+    }
```

Comment:
`isInvalidFieldValue` method has a redundant invocation of `isInvalidFieldReference`
I think this can be removed, as below  (line 82) you are explicitly invoking the `isInvalidFieldReference`

Reply:
I was using `isInvalidFieldValue` independently from `isInvalidFieldReference` but then ended up with `validateFieldReference` and its if statements. It's kind of redundant just for the case of `validateFieldReference` but I'll remove it.